File: app/src/main/java/com/bhimanshukalra/studentmanagementappfragments/Adapter/StudentListAdapter.java
package com.bhimanshukalra.studentmanagementappfragments.Adapter;

import android.support.annotation.NonNull;
import android.support.v7.widget.CardView;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.bhimanshukalra.studentmanagementappfragments.R;
import com.bhimanshukalra.studentmanagementappfragments.model.Student;

import java.util.ArrayList;

/**
 * The Student list RecyclerView adapter.
 */
public class StudentListAdapter extends RecyclerView.Adapter<StudentListAdapter.RecyclerViewHolder> {
    private static ArrayList<Student> mStudentList;
    private static RecyclerClickListener mClickListener;

    /**
     * Instantiates a new Student list adapter.
     *
     * @param clickListener the click listener of list item.
     */
    public StudentListAdapter(RecyclerClickListener clickListener) {
        mClickListener = clickListener;
    }

    /**
     * Set data in recycler.
     *
     * @param studentList the list of student details.
     */
    public static void setDataInRecycler(ArrayList<Student> studentList) {
        mStudentList = studentList;
    }

    @NonNull
    @Override
    public RecyclerViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        CardView cardView = (CardView) LayoutInflater.from(viewGroup.getContext()).inflate(R.layout.item_student, viewGroup, false);
        return new RecyclerViewHolder(cardView);
    }

    @Override
    public void onBindViewHolder(@NonNull RecyclerViewHolder recyclerViewHolder, int position) {
        View view = recyclerViewHolder.mCardView;
        TextView tvName = view.findViewById(R.id.item_student_tv_name);
        TextView tvClass = view.findViewById(R.id.item_student_tv_class);
        TextView tvRollNumber = view.findViewById(R.id.item_student_tv_roll_num);

        tvName.setText(mStudentList.get(position).getName());
        tvClass.setText(mStudentList.get(position).getClassName());
        tvRollNumber.setText(String.valueOf(mStudentList.get(position).getRollNumber()));
    }

    @Override
    public int getItemCount() {
        if (mStudentList != null) {
            return mStudentList.size();
        }
        return 0;
    }

    /**
     * The interface Recycler click listener.
     */
    public interface RecyclerClickListener {
        /**
         * List item clicked.
         *
         * @param view     the view clicked on.
         * @param position the position of clicked item in list.
         */
        void listItemClicked(View view, int position);
    }

    /**
     * The type Recycler view holder.
     */
    public class RecyclerViewHolder extends RecyclerView.ViewHolder {
        private CardView mCardView;

        /**
         * Instantiates a new Recycler view holder.
         *
         * @param cardView the card view showing all the details of individual students.
         */
        public RecyclerViewHolder(@NonNull CardView cardView) {
            super(cardView);
            mCardView = cardView;
            mCardView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View view) {
                    mClickListener.listItemClicked(view, getLayoutPosition());
                }
            });
        }
    }
}

File: app/src/main/java/com/bhimanshukalra/viewpagerkotlin/models/StudentListAdapter.kt
package com.bhimanshukalra.viewpagerkotlin.models

import android.view.LayoutInflater
import android.view.ViewGroup
import android.widget.TextView
import androidx.cardview.widget.CardView
import androidx.recyclerview.widget.RecyclerView
import com.bhimanshukalra.viewpagerkotlin.R

class StudentListAdapter(
    private var mStudentList: List<Student>,
    private var mClickListener: RecyclerClickListener
) : RecyclerView.Adapter<StudentListAdapter.RecyclerViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerViewHolder {
        val cardView: CardView = LayoutInflater.from(parent.context)
            .inflate(R.layout.item_student, parent, false) as CardView
        return RecyclerViewHolder(cardView)
    }

    override fun getItemCount(): Int {
        return mStudentList.size
    }

    override fun onBindViewHolder(holder: RecyclerViewHolder, position: Int) {
        holder.tvName.text = mStudentList.get(position).name
        holder.tvClass.text = mStudentList.get(position).className
        holder.tvRollNumber.text = mStudentList.get(position).rollNumber.toString()
    }

    inner class RecyclerViewHolder(val cardView: CardView) :
        RecyclerView.ViewHolder(cardView) {
        var tvName: TextView = cardView.findViewById(R.id.item_student_tv_name)
        var tvClass: TextView = cardView.findViewById(R.id.item_student_tv_class)
        var tvRollNumber: TextView = cardView.findViewById(R.id.item_student_tv_roll_num)

        init {
            cardView.setOnClickListener {
                mClickListener.listItemClicked(adapterPosition)
            }
        }
    }

    interface RecyclerClickListener {
        fun listItemClicked(position: Int)
    }
}




















